src/claimAttest/claimAttest.service.ts

```diff
@@ -171,10 +170,138 @@ export class ClaimAttestService {
       });
     }
   }
-  public search() {
-    
+
+  public async search(token: string, response) {
+    if (token) {
+      const searchUser = await this.keycloakService.getUserTokenAccount(token);
+
+      if (searchUser?.error) {
+        return response.status(401).send({
+          success: false,
+          status: 'keycloak_token_bad_request',
+          message: 'You do not have access for this request.',
+          result: null,
+        });
+      } else if (!searchUser?.username) {
+        return response.status(401).send({
+          success: false,
+          status: 'keycloak_token_error',
+          message: 'Your Login Session Expired.',
+          result: null,
+        });
+      } else {
+        let name = '';
+        let dob = '';
+        let gender = '';
+        if (
+          searchUser?.attributes?.gender &&
+          searchUser?.attributes?.dob &&
+          searchUser?.attributes?.name
+        ) {
+          //login with digilocker
+          name = searchUser?.attributes?.name[0];
+          dob = await this.convertDate(searchUser?.attributes?.dob[0]);
+          gender = searchUser?.attributes?.gender[0];
+        } else {
+          //login with mobile and otp
+          const sb_rc_search = await this.sbrcService.sbrcSearchEL(
+            'Instructor',
+            {
+              filters: {
+                username: {
+                  eq: searchUser?.username,
+                },
+              },
+            },
+          );
+
+          if (sb_rc_search?.error) {
+            return response.status(501).send({
+              success: false,
+              status: 'sb_rc_search_error',
+              message: 'System Search Error ! Please try again.',
+              result: sb_rc_search?.error.message,
+            });
+          } else if (sb_rc_search.length === 0) {
+            return response.status(404).send({
+              success: false,
+              status: 'sb_rc_search_no_found',
+              message: 'Data Not Found in System.',
+              result: null,
+            });
+          } else {
+            name = sb_rc_search[0].name;
+            dob = sb_rc_search[0].dob;
+            gender = sb_rc_search[0].gender;
+          }
+        }
+
+        //search count
+        let searchSchema = {
+          filters: {
+            name: {
+              eq: name,
+            },
+            dob: {
+              eq: dob,
+            },
+            gender: {
+              eq: gender,
+            },
+          },
+        };
+
+        const userDetails = await this.sbrcService.sbrcSearch(
+          searchSchema,
+          'Instructor',
+        );
+        if (userDetails[0]?.school_id) {
```

Comment:
@YoginiTayade  Add else part of this if 
in else part add "school id not found in instructor account" with status: false 400

Reply:
resolved